subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -173,27 +190,34 @@ private String getFilename(Path dir) {
 
         @Override
         public FileVisitResult visitFile(Path file, @Nullable BasicFileAttributes attrs) {
-            String name = stringInterner.intern(file.getFileName().toString());
+            String name = intern(file.getFileName().toString());
             if (isAllowed(file, name, false, attrs, builder.getRelativePath())) {
-                if (attrs == null) {
-                    throw new FileSnapshottingException(String.format("Cannot read file '%s': not authorized.", file));
-                }
-                if (attrs.isSymbolicLink()) {
-                    // when FileVisitOption.FOLLOW_LINKS, we only get here when link couldn't be followed
-                    throw new FileSnapshottingException(String.format("Could not list contents of '%s'. Couldn't follow symbolic link.", file));
-                }
-                addFileSnapshot(file, name, attrs);
+                FileSystemLocationSnapshot snapshot = getSnapshot(file, attrs, name);
+                builder.visit(snapshot);
             }
             return FileVisitResult.CONTINUE;
         }
 
+        private FileSystemLocationSnapshot getSnapshot(Path file, BasicFileAttributes attrs, String name) {
+            try {
+                String absolutePath = intern(file.toString());
+                HashCode hash = hasher.hash(file.toFile(), attrs.size(), attrs.lastModifiedTime().toMillis());
+                FileMetadata fileMetadata = FileMetadata.from(attrs);
+                return new RegularFileSnapshot(absolutePath, name, hash, fileMetadata);
+            } catch (UncheckedIOException e) {
+                LOGGER.debug("Could not read file path '{}'.", file, e);
```

Comment:
I think these log statements should at least be at the `info` level if not at the `warn` level. I think it is interesting to the user that something strange is going on in the snapshotted directory. WDYT?

Reply:
had `warn` previously, @lptr explicitly requested `debug` here

Reply:
@lptr, @wolfs, please decide.

Reply:
I think we need to decide on a pattern for using `WARN` in general, or better yet, what is the role of each level.

In many places `WARN` is used as a stand-in for `LIFECYCLE`, and used to print regular messages about the progression of the build. We do use it to show actual warnings in other places, though.

The underlying problem is that we have no good way to report warnings for a number of reasons. What I'm afraid here is if we add a `WARN` message then we'll just further clutter the output without giving any value.

I guess finding a broken symlink is not an error condition anymore, so for me `DEBUG` seems okay.

Reply:
I'm okay with that, what do you think @wolfs?

Reply:
I wonder why we don't use `info`? Do we not want to use `info` any more? What is the `INFO` loglevel for?

Reply:
@lptr?

Reply:
Changed it to `info`

Reply:
I don't care enough to fight for this, I think `INFO` is fine.

Reply:
I really wanted to find out if we have some guidelines when to use `INFO` and when to use `DEBUG`, since I am struggling with that question myself. Who is the intended audience of `INFO`?